datatree/io.py

```diff
@@ -68,16 +68,8 @@ def _open_datatree_netcdf(filename: str, **kwargs) -> DataTree:
         tree_root = DataTree.from_dict({"/": ds})
         for path in _iter_nc_groups(ncds):
             subgroup_ds = open_dataset(filename, group=path, **kwargs)
+            tree_root[path] = DataTree(name=path, data=subgroup_ds)
```

Comment:
Using `__setitem__` instead of `_set_item` here can behave differently, for instance if a group appeared twice one would allow overwriting by default and the other wouldn't.

Reply:
The reason I made the `_set_item_` method was for internal developer API so that this kind of behaviour could be controlled more explicitly.

Reply:
Thanks for the explanation.